datanode/candlesv2/candle_updates.go

```diff
@@ -198,7 +190,7 @@ func (s *CandleUpdates) getCandleUpdates(ctx context.Context, lastCandle *entiti
 	if lastCandle != nil {
 		start := lastCandle.PeriodStart
 		var candles []entities.Candle
-		candles, _, err = s.candleSource.GetCandleDataForTimeSpan(ctx, s.candleID, &start, nil, entities.CursorPagination{})
+		candles, _, err = s.candleSource.GetCandleDataForTimeSpan(ctx, s.candleID, &start, &now, entities.CursorPagination{})
 
 		if err != nil {
 			return nil, fmt.Errorf("getting candle updates:%w", err)
```

Comment:
Theres a little bit below here:
```
for _, candle := range candles {
	if candle.LastUpdateInPeriod.After(lastCandle.LastUpdateInPeriod) {
		updates = append(updates, candle)
	}
}
```

which I think will stop the empty candles being added because for an empty candle, for whatever reason `LastUpdateInPeriod` ends up being equal to the value of `LastUpdateInPeriod` for the last real candle.

Reply:
You're right, I'll see if I can replace the `locf` in the subquery with an `interpolate` call. I think ought to set the last update in period to the correct value.